src/NodaTime/Calendars/Wondrous/WondrousCalendarHelper.cs

```diff
@@ -0,0 +1,2204 @@
+using JetBrains.Annotations;
+using NodaTime.Utility;
+using System;
+using System.Collections.Generic;
+
+namespace NodaTime.Calendars.Wondrous
+{
+    /// <summary>
+    /// Helper and extension methods for the Wondrous Calendar.
+    /// </summary>
+    /// <remarks>
+    /// The Wondrous Calendar Helper is in a distinct namespace so that the Extensions are only available when this
+    /// namespace is included.
+    /// </remarks>
+    public static class WondrousCalendarHelper
+    {
+        private const int Month18 = 18;
+        private const int DaysInMonth = 19;
+
+
+        /// <remarks>
+        ///     There are 19 months in a year. Between the 18th and 19th month are the "days of Ha" (Ayyam-i-Ha).
+        ///     Options for numbering the months:
+        ///     - treat Ayyam-i-Ha as month 0. This will cause problems if 0 is treated as unknown. When stored internally,
+        ///       NodaTime months are converted to 0-based numbers, so this would by -1 which is not supported.
+        ///     - treat Ayyam-i-Ha as month -1. This would be a problem if the base classes convert to 0 based months
+        ///     - treat Ayyam-i-Ha as month 19. This is confusing because the 19th month would be displayed as month #20.
+        ///     - treat Ayyam-i-Ha as month 20. This will cause problems if months are sorted naively.
+        ///     - treat Ayyam-i-Ha as extra days in month 18. This would require special handling to sometimes display
+        ///       days in "month 18" as being Ayyam-i-Ha, but we have no direct control over display.
+        ///       
+        /// This version will treat Ayyam-i-Ha days as extra days in month 18.
+        /// 
+        /// Users interacting with the system should use 0 as Ayyam-i-Ha.
+        /// </remarks>
+        internal const int AyyamiHaMonth0 = 0;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="wondrousYear">Year in the Wondrous calendar</param>
```

Comment:
I'd rename these to just `year`, `month`, `date` - but *possibly* rename the method to `CreateWondrousDate` as then if the helper is imported statically, it can be called very easily.

Reply:
Sounds good.